yb-voyager/src/query/queryissue/parser_issue_detector.go

```diff
@@ -472,6 +473,55 @@ func (p *ParserIssueDetector) genericIssues(query string) ([]QueryIssue, error)
 	return result, nil
 }
 
+func (p *ParserIssueDetector) GetRedundantIndexIssues(redundantIndexes []utils.RedundantIndexesInfo) []QueryIssue {
+
+	redundantIndexToInfo := make(map[string]utils.RedundantIndexesInfo)
+
+	//This function helps in resolving the existing index in cases where existing index is also a redundant index on some other index
+	//So in such cases we need to report the main existing index.
+	/*
+		e.g. INDEX idx1 on t(id); INDEX idx2 on t(id, id1); INDEX idx3 on t(id, id1,id2);
+		redundant index coming from the script can have
+		Redundant - idx1, Existing idx2
+		Redundant - idx2, Existing idx3
+		So in this case we need to report it like
+		Redundant - idx1, Existing idx3
+		Redundant - idx2, Existing idx3
+	*/
+	resolvedRedundantIndexInfo := func(currRedundantIndexInfo utils.RedundantIndexesInfo) utils.RedundantIndexesInfo {
```

Comment:
I'm thinking  - might be better to phrase this as `getRootExistingIndex` ? idk, "resolved" isn't very clear immediately.